specs/tables.md

```diff
@@ -356,3 +356,72 @@ The table below lists all of copy pairs supported in the copy table:
 |        |         |        |           |            |                |                |            |        | $rlcAcc |         |     |           |                 |
 | 1      | 0/1     | 0      | $callID   | Memory     | $memoryAddress | $memoryAddress | $bytesLeft | $value | $rlcAcc | -       | 0/1 | $counter  | $rwcIncLeft     |
 | 0      | 0       | 0/1    | $txID     | TxLog      | $byteIndex \|\| TxLogData \|\| $logID | - | - | $value | $rlcAcc | -       | 0   | $counter  | $rwcIncLeft     |
+
+## Exponentiation Table
+
+Proved by the Exponentiation circuit.
+
+The exponentiation table is a virtual table within the exponentiation circuit assignments. An exponentiation operation `a ^ b == c (mod 2^256)` is broken down into steps that perform the exponentiation by squaring.
+
+The following algorithm is used for exponentiation by squaring:
+```
+Function exp_by_squaring(x, n)
+    if n = 0  then return  1;
+    if n = 1  then return  x;
+    if n is odd:
+	return x * exp_by_squaring(x, n - 1)
+    if n is even:
+	return (exp_by_squaring(x, n / 2))^2
+```
+
+Using the above algorithm, `3 ^ 13 == 1594323 (mod 2^256)` is broken down into the following steps:
+```
+3      * 3   = 9
+9      * 3   = 27
+27     * 27  = 729
+729    * 729 = 531441
+531441 * 3   = 1594323
+```
+
+We assign the above steps to the exponentiation table in the reverse order, so that the first step is `531441 * 3 = 1594323`. From here on, the RHS in the above steps is termed as `intermediate_exponentiation`. We define another term `intermediate_exponent` as a value that starts at the integer exponent of the operation, i.e. `13` in the above case, and reduces down to `2` such that:
```

Comment:
Why do we have to do the reverse order? It seems possible to do the normal order and add up to the exponent.

Reply:
It can be done in the normal order as well. When I started designing this initially, it was more intuitive for me to do a lookup to the first step of the trace, instead of the last step.

It was just a design choice from my side, it can in fact be done the other way as well. What do you think?

Reply:
Reverse order sounds great to me.